### PR TITLE
WDIO video recording

### PR DESCRIPTION
**Pull request by AronNovak**
Current status:
 - possible to launch docker-compose

sel

Things to do:
 - [x] test local video recording
 - [x] upload video
- [ ] GitHub comment

#83

### CODE REVIEW

**Comment by AronNovak** [comment]
Todo:
set test name via `desiredCapabilities.setCapability("name", "myTestName");` - https://github.com/zalando/zalenium/blob/master/docs/usage_examples.md  - to have meaningful video name

**Comment by AronNovak** [comment]
Now the filename contains the spec name:
filename

**Comment by AronNovak** [comment]
As the whole flow is not 100% ready, I made the robot comment tested via a locally crafted shell script, welcome @gizra-robot
```
GH_COMMENT=/tmp/video_urls
cat /dev/null > $GH_COMMENT
echo -n '{
  "body": "' >> $GH_COMMENT
echo -n "Test comment by the robot" >> $GH_COMMENT
echo '"}' >> $GH_COMMENT
cat $GH_COMMENT
curl -H "Authorization: token $GH_TOKEN" --data @$GH_COMMENT https://api.github.com/repos/Gizra/drupal-elm-starter/issues/84/comments
```

**Comment by AronNovak** [comment]
@amitaibu I exhausted the timebox, as you see all the components are working, but not together seamlessly. I think from now, +2 hours would make it happen. Now the only realistic environment to test is Travis, to get the branch name, and so on.

**Comment by amitaibu** [comment]
Hi there @gizra-robot 😄 

I've bumped by 2h - Looks really good so far!

**Comment by AronNovak** [comment]
@gizra-robot Thanks for the comment, my dear, the test should not be green however, everything else seems to be smooth!

**Comment by amitaibu** [comment]
@gizra-robot  (and @AronNovak 😉 ) - could you change `The video of the failed test case is available from` => `* Video of the failed WebdriverIO test [login.js](link-to-GDrive).`

**Comment by amitaibu** [comment]
(^^ can be follow up)

**Comment by AronNovak** [comment]
We have one broken build (i fought with not having good exit status, so it"s important), we have a passing build after I reverted the intentional broke of the test, so it seems to be nearly ready.
There are few things to keep in mind:
 - repo URL must be rewritten if transferred to another project (https://github.com/Gizra/drupal-elm-starter/pull/84/files#diff-be525526252e2d662882d9ca5560a7cdR78), possibly it could be exposed into some high-level config
 - no automatic cleanup of Google Drive, one day it will hit the quota

File: ci-scripts/test_server.sh [hunk 1]
```diff
@@ -8,25 +8,82 @@ set -e
...
+      URL=$(/tmp/gdrive info --service-account gdrive-service-account.json $ID  | grep ViewUrl | sed s/ViewUrl\:\ //)
+      echo -n "* Video of the failed WebdriverIO test [$FAILED_SPEC]($URL)." | tee -a $GH_COMMENT
+      echo ""
+      echo -n "\n" >> $GH_COMMENT
+
+    done;
+    echo '"}' >> $GH_COMMENT
+
+    # Todo: make it non-Gizra specific by detecting the user of the repository.
+    PR_URL=$(curl -H "Authorization: token $GH_TOKEN" -s  https://api.github.com/repos/Gizra/drupal-elm-starter/pulls?head=Gizra:${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH} | grep '"url"' | head -n1  | cut -d'"' -f 4)
```
Comment (amitaibu) [inline ci-scripts/test_server.sh]: maybe the repo name should be part of the [travis.config.sh](https://github.com/Gizra/drupal-elm-starter/blob/master/server/travis.config.sh)?

**Comment by amitaibu** [comment]
We seem to get the same test uploaded twice (as in the [comment](https://github.com/Gizra/drupal-elm-starter/pull/84#issuecomment-294119056))

**Comment by AronNovak** [comment]
The thing is that actually we should three 3 files uploaded for one failed spec, as we execute it three times. We can say that we only upload one file and assume that it failed the same way if it failed always however.

**Comment by amitaibu** [comment]
Yeah, that sounds right. 1 video per 1 fail

File: ci-scripts/test_server.sh (hunk 1 above)
Comment (AronNovak) [inline ci-scripts/test_server.sh]: Yep, thanks, implemented.

File: server/travis.config.sh [hunk 2]
```diff
@@ -8,6 +8,9 @@
 #
 #########################################################################################
 
+# GitHub Repository identifier
+GH_REPO="Gizra/drupal-elm-starter"
```
Comment (amitaibu) [inline server/travis.config.sh]: 👍 lets also mention in the README as part of the installation notes

Comment (AronNovak) [inline server/travis.config.sh]: Thanks, README is extended, also with the description of Google Drive credentials. I just try to make the tests fail one more time to test the changes in the upload process.

**Comment by AronNovak** [comment]
Ready for (a final) review, the failed + successful test pair from Travis is there, suggestions are implemented.

File: server/README.md [hunk 3]
```diff
@@ -97,3 +98,15 @@ when you run the upgrade script.
...
 7. Restore the backup of the sites/default folder.
+
+## Start a new project based on Drupal Elm Starter
+
+Edit `server/travis.config.sh` and actualize `GH_REPO` variable with the new
+location of the project.
+WDIO test failures are automatically uploaded to Google Drive. The credentials
+in this public repository are encrypted by Travis. If you don't fork the
+project, but copy it, you need to re-add and encrypt your credentials,
+the process of updating `gdrive-service-account.json.enc` is described at
```
Comment (amitaibu) [inline server/README.md]: these are long docs. Is there a tl;dr version of it?

**Comment by AronNovak** [comment]
Sure, the README is now extended and restructured.

**Comment by amitaibu** [comment]
Wonderful, thanks!

**Comment by DavidHernandez** [comment]
Nice work!

**Pull request merged**